pkg/uwb-dw1000/hal/uwb_dw1000.c

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (C) 2020 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_uwb_dw1000
+ * @{
+ *
+ * @file
+ * @brief       Glue code for running uwb-core for RIOT
+ * *
+ * @}
+ */
+
+#include "dpl/dpl.h"
+#include "dw1000/dw1000_phy.h"
+#include "dw1000/dw1000_hal.h"
+#include "uwb_dw1000.h"
+
+#include "uwb/uwb.h"
+#include "uwb/uwb_mac.h"
+
+#include "utlist.h"
+#include "thread.h"
+
+#ifndef LOG_LEVEL
+#define LOG_LEVEL   LOG_NONE
+#endif
+#include "log.h"
+
+#ifndef DW1000_THREAD_PRIORITY
+#define DW1000_THREAD_PRIORITY  MYNEWT_VAL_UWB_DEV_TASK_PRIO
+#endif
+
+#ifndef DW1000_ROLE_DEFAULT
+#define DW1000_ROLE_DEFAULT     0x0
+#endif
+
+#ifndef DW1000_RX_DIAGNOSTIC
+#define DW1000_RX_DIAGNOSTIC    0
+#endif
+
+/* Link list head */
+static struct _dw1000_dev_instance_t * dw1000_instance_head;
+
+/* Default instance configuration */
+static const dw1000_dev_instance_t dw1000_instance_config_default = {
```

Comment:
There are lots of hardcoded default values in this struct. Do you think it would make sense to allow them to be overridable (like `DW1000_ROLE_DEFAULT`) ?

Reply:
OK.

Reply:
done